internal/backends/collection.go

```diff
@@ -71,6 +71,7 @@ func CollectionContract(c Collection) Collection {
 type QueryParams struct {
 	// TODO https://github.com/FerretDB/FerretDB/issues/3235
 	Filter *types.Document
+	Sort   *types.Document
```

Comment:
Rather than passing not validated document, we could validate `*types.Document` in handler and then pass a struct here like it's done for Index. Then we won't need to return `NewCommandErrorMsgWithArgument` from postgresql implementation. :hugs: